### PR TITLE
salt-lint: add 0.9.2

### PR DESCRIPTION
**Pull request by NoRePercussions**
## Description of changes

Add the [`salt-lint` python application](https://github.com/warpnet/salt-lint) v0.9.2.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] **(See below)** Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Note: I had trouble getting pytest to find any tests. CDing into `$out` did not work, nor did pointing pytest specifically at `tests/unit`

### CODE REVIEW

File: pkgs/by-name/sa/salt-lint/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ];
+
+  pythonImportsCheck = [ "saltlint" ];
+
+  passthru.tests = {
+    version = testers.testVersion { package = salt-lint; };
+  };
+
+  meta = with lib; {
+    description = "A command-line utility that checks for best practices in SaltStack.";
```
Comment (Aleksanaa) [inline pkgs/by-name/sa/salt-lint/package.nix:25]:
```suggestion
    description = "A command-line utility that checks for best practices in SaltStack";
```

File: pkgs/by-name/sa/salt-lint/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+python3Packages.buildPythonApplication rec {
+  pname = "salt-lint";
+  version = "0.9.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-f3Tmguf9eHIqbTkeqO3J/HlRE+z9QGV9aAV9QE7nvo4=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (Aleksanaa) [inline pkgs/by-name/sa/salt-lint/package.nix:12]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/sa/salt-lint/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  pythonImportsCheck = [ "saltlint" ];
+
+  passthru.tests = {
+    version = testers.testVersion { package = salt-lint; };
+  };
+
+  meta = with lib; {
+    description = "A command-line utility that checks for best practices in SaltStack.";
+    homepage = "https://github.com/warpnet/salt-lint";
```
Comment (Aleksanaa) [inline pkgs/by-name/sa/salt-lint/package.nix:26]:
```suggestion
    homepage = "https://salt-lint.readthedocs.io";
```

This page has more information.

**Comment by NoRePercussions** [comment]
Hi, bumping this -- any chance it will be merged soon?

**Review by MaxwellDupre — APPROVED** [review]
LGTM

File: pkgs/by-name/sa/salt-lint/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  ];
+
+  pythonImportsCheck = [ "saltlint" ];
+
+  passthru.tests = {
+    version = testers.testVersion { package = salt-lint; };
+  };
+
+  meta = with lib; {
+    description = "A command-line utility that checks for best practices in SaltStack";
```
Comment (Aleksanaa) [inline pkgs/by-name/sa/salt-lint/package.nix:25]:
```suggestion
    description = "Command-line utility that checks for best practices in SaltStack";
```

**Review by Aleksanaa — COMMENTED** [review]
Looks okay. Please reword to `maintainers: add norepercussions` and `salt-lint: init at 0.9.2`.

File: pkgs/by-name/sa/salt-lint/package.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "salt-lint";
+  version = "0.9.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-f3Tmguf9eHIqbTkeqO3J/HlRE+z9QGV9aAV9QE7nvo4=";
+  };
+
+  dependencies = with python3Packages; [
+    pipBuildHook
```
Comment (Aleksanaa) [inline pkgs/by-name/sa/salt-lint/package.nix:13]: Looks like this should be in `build-system`?

File: pkgs/by-name/sa/salt-lint/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/sa/salt-lint/package.nix:1]: Your formatter doesn't seem to be nixfmt-rfc-style. Try to reformat it.

**Review by Aleksanaa — COMMENTED** [review]
Others lgtm

File: pkgs/by-name/sa/salt-lint/package.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  fetchPypi,
+  python3Packages,
+  testers,
+  salt-lint,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "salt-lint";
+  version = "0.9.2";
```
Comment (SuperSandro2000) [inline pkgs/by-name/sa/salt-lint/package.nix:10]:
```suggestion
  version = "0.9.2";
  pyproject = true;
```

**Comment by Aleksanaa** [comment]
@ofborg eval

**Pull request closed without merging**